kernel/exports/library.c

```diff
@@ -213,28 +214,36 @@ klibrary_t * library_lookup(const char * name) {
     return lib;
 }
 
-klibrary_t * library_open(const char * name, const char * fn) {
-    klibrary_t * lib;
+klibrary_t *library_lookup_fn(const char *fn) {
+    klibrary_t *lib;
+
+    irq_disable_scoped();
+
+    LIST_FOREACH(lib, &library_list, list) {
+        if(!strncmp(lib->image.fn, fn, NAME_MAX))
+            break;
+    }
+
+    if(!lib)
+        errno = ENOENT;
+
+    return lib;
+}
+
+klibrary_t *library_open(const char *name, const char *fn) {
+    klibrary_t *lib = NULL;
 
     // If they passed us a valid name, try that first.
     if(name) {
         lib = library_lookup(name);
+    }
+    else {
+        lib = library_lookup_fn(fn);
+    }
 
-        if(lib) {
-            // Make sure lib_open is valid. Note that since we
-            // _have_ found something, any inconsistencies here
-            // are errors and not a signal to load another copy.
-            if(!lib->lib_open) {
-                errno = EINVAL;
-                return NULL;
-            }
```

Comment:
This is an unnecessary condition in general. Because if `lib_open` function is not there, then the elf file itself will not load. I don’t understand how it can disappear if you got loaded library.